packages/components/src/components/DistributionChart.tsx

```diff
@@ -6,40 +6,46 @@ import { distributionErrorToString } from "@quri/squiggle-lang";
 import { createClassFromSpec } from "react-vega";
 import * as chartSpecification from "../vega-specs/spec-distributions.json";
 import { ErrorBox } from "./ErrorBox";
-import styled from "styled-components";
+import { useSize } from "react-use";
 
 let SquiggleVegaChart = createClassFromSpec({
   spec: chartSpecification as Spec,
 });
 
 type DistributionChartProps = {
   distribution: Distribution;
-  width: number;
+  width?: number;
   height: number;
 };
 
 export const DistributionChart: React.FC<DistributionChartProps> = ({
   distribution,
-  width,
   height,
+  width,
 }: DistributionChartProps) => {
-  let shape = distribution.pointSet();
-  if (shape.tag === "Ok") {
-    let widthProp = width ? width - 20 : undefined;
-    var result = (
-      <SquiggleVegaChart
-        data={{ con: shape.value.continuous, dis: shape.value.discrete }}
-        width={widthProp}
-        height={height}
-        actions={false}
-      />
-    );
-  } else {
-    var result = (
-      <ErrorBox heading="Distribution Error">
-        {distributionErrorToString(shape.value)}
-      </ErrorBox>
-    );
-  }
-  return result;
+  const [sized, _] = useSize((size) => {
+    let shape = distribution.pointSet();
+    let widthProp = width !== undefined ? width - 20 : size.width - 10;
```

Comment:
The - 10 here is a bit of a magic number, it prevents the graph from going off into infinite size, because the padding for this is 5 in each direction. I think?

Reply:
I'm pretty sure. I think I probably previously messed with it until it stopped crashing, which was around at 10.